CASMPET-6898 update CSM only upgrade to set CFS on storage nodes (#4636)
Cray-HPE/docs-csm: upgrade/Stage_0_Prerequisites.md

   +-----------------------------+--------------------------------------+--------------------------------------+-----------------------------+----------------------------+
   | master-secure-kubernetes    | c1bcaf00-109d-470f-b665-e7b37dedb62f | a22fb912-22be-449b-a51b-081af2d7aff6 | management-22.4.0-csm-x.y.z | Management_Master          |
   | worker-secure-kubernetes    | 8b1343c4-1c39-4389-96cb-ccb2b7fb4305 | 241822c3-c7dd-44f8-98ca-0e7c7c6426d5 | management-22.4.0-csm-x.y.z | Management_Worker          |
   | storage-secure-storage-ceph | f3dd7492-c4e5-4bb2-9f6f-8cfc9f60526c | 79ab3d85-274d-4d01-9e2b-7c25f7e108ca | storage-22.4.0-csm-x.y.z    | Management_Storage         |
   +-----------------------------+--------------------------------------+--------------------------------------+-----------------------------+----------------------------+
   ```

>    Save the name of the CFS configuration:

Reviewer: ```suggestion
   Save the name of the CFS configurations:
```

   ```bash

>    export CFS_CONFIG_NAME="management-22.4.0-csm-x.y.z"

Reviewer: For consistency with `STORAGE_CFS_CONFIG_NAME`, it might be nice to name this `MGMT_CFS_CONFIG_NAME` or `KUBERNETES_CFS_CONFIG_NAME`.

```suggestion
   export KUBERNETES_CFS_CONFIG_NAME="management-22.4.0-csm-x.y.z"
```

   disables the components and does not clear their configuration states or error counts. When the
   nodes are rebooted to their new images later in the CSM upgrade, they will automatically be
   enabled in CFS, and node personalization will occur.
  

>     1. `(ncn-m001)` Update CFS on worker nodes and master nodes.

Reviewer: ```suggestion
    1. `(ncn-m001)` Apply the appropriate CFS configuration to worker nodes and master nodes.
```

        Get master nodes' and worker nodes' xnames.

        ```bash

>         XNAME_WORKERS=$(cray hsm state components list --role Management --subrole Worker --type Node --format json |
>           jq -r '.Components | map(.ID) | join(",")')
>         XNAME_MASTERS=$(cray hsm state components list --role Management --subrole Master --type Node --format json |
>           jq -r '.Components | map(.ID) | join(",")')
>         echo "${XNAME_MASTERS},${XNAME_WORKERS}"

Reviewer: super nitpicky, but `WORKER_XNAMES` and `MASTER_XNAMES` more accurately describes what the variable holds.

```suggestion
        WORKER_XNAMES=$(cray hsm state components list --role Management --subrole Worker --type Node --format json |
          jq -r '.Components | map(.ID) | join(",")')
        MASTER_XNAMES=$(cray hsm state components list --role Management --subrole Master --type Node --format json |
          jq -r '.Components | map(.ID) | join(",")')
        echo "${MASTER_XNAMES},${WORKER_XNAMES}"
```

        ```

>         Set CFS on master nodes and worker nodes using the xnames found in the previous step.

Reviewer: ```suggestion
        Apply the CFS configuration to master nodes and worker nodes using the xnames and CFS configuration name found in the previous steps.
```

        ```bash
        /usr/share/doc/csm/scripts/operations/configuration/apply_csm_configuration.sh \

>             --no-config-change --config-name "${CFS_CONFIG_NAME}" --no-enable --no-clear-err \

Reviewer: If you do the `CFS_CONFIG_NAME` variable rename suggested above.

```suggestion
            --no-config-change --config-name "${KUBERNETES_CFS_CONFIG_NAME}" --no-enable --no-clear-err \
```

>             --xnames ${XNAME_MASTERS},${XNAME_WORKERS}

Reviewer: ```suggestion
            --xnames ${MASTER_XNAMES},${WORKER_XNAMES}
```

        ```

        Successful output will end with the following:

        ```text
        All components updated successfully.
        ```

>     1. `(ncn-m001)` Update CFS on storage nodes.

Reviewer: ```suggestion
    1. `(ncn-m001)` Apply the appropriate CFS configuration to storage nodes.
```

        Get storage nodes' xnames.

        ```bash

>         XNAME_STORAGE=$(cray hsm state components list --role Management --subrole Storage --type Node --format json |

Reviewer: ```suggestion
        STORAGE_XNAMES=$(cray hsm state components list --role Management --subrole Storage --type Node --format json |
```

          jq -r '.Components | map(.ID) | join(",")')

>         echo $XNAME_STORAGE

Reviewer: ```suggestion
        echo $STORAGE_XNAMES
```

        ```

>         Set CFS on storage nodes using the xnames found in the previous step.

Reviewer: ```suggestion
        Apply the CFS configuration to storage nodes using the xnames and CFS configuration name found in the previous steps.
```

        ```bash
        /usr/share/doc/csm/scripts/operations/configuration/apply_csm_configuration.sh \
            --no-config-change --config-name "${STORAGE_CFS_CONFIG_NAME}" --no-enable --no-clear-err \

>             --xnames ${XNAME_STORAGE}

Reviewer: ```suggestion
            --xnames ${STORAGE_XNAMES}
```
